elements/lisk-p2p/src/peer_directory/new_list.ts

```diff
@@ -50,18 +50,15 @@ export class NewList extends PeerList {
 	}
 
 	// Extend eviction of NewPeers
-	public evictPeer(bucketId: number): CustomPeerInfo {
+	public evictPeer(bucketId: number): CustomPeerInfo | undefined {
```

Comment:
This `undefined` doesn't seem ok to me. We should always evict at least a peer when we call this method, as per its name states. I suggest to split the action in two:
- when adding a peer, pick the bucket and check if it's full
-  if it's full, then evict a peer or several

Reply:
Yes, this function should always evict some peer to make space for the new one. For `newPeers`, we have two strategies and the second strategy will always return one evictedPeer. So overall, this function should always return one evictedPeer